public/src/app.js

```diff
@@ -8,9 +8,11 @@ import { initUsername } from "./utils/initUsername.js";
 import { chatMessages } from "./components/chatMessages.js";
 
 const app = (s) => {
-  const socket = io.connect("http://localhost:3000");
+  let socket;
 
   s.setup = function () {
+    socket = io.connect("http://localhost:3000");
```

Comment:
I suspect the race condition had to do w/ initializing the socket connection _outside_ of the setup method  - doing so within (where all the socket listeners that depend on it live) seems to resolve the issue.